doc/wrapping_js_components.md

```diff
@@ -0,0 +1,693 @@
+  aForwardedRef: React.ForwardedRef<any>;
+}
+```
+
+Refs are special JS object types that need to be converted to and from Dart using interop utilities. Converting any one of those ref types looks like:
+
+```dart
+@Props(keyNamespace: '')
+mixin ArbitraryComponentProps on UiProps {
+  dynamic get someRef => unjsifyRefProp(_someRef$rawJs);
+
+  set someRef(dynamic value) => _someRef$rawJs = jsifyRefProp(value);
+
+  @Accessor(key: 'someRef')
+  dynamic _someRef$rawJs;
+}
+```
+
+> **For more information** on why getters and setters are used here, see the [Conversion through Getters and Setters section](#conversion-through-getters-and-setters).
+
+The steps are:
+
+1. Create a `dynamic` property that will represent the raw JS ref property for the component. This is necessary because the raw JS is not in a consumable form and should not be accessed directly.
+1. Add an `@Accessor` annotation to the property with a key that matches the prop name. This will link the Dart ref property to the JS prop.
+1. Add ref getters and setters named the same as the prop.
+   - The getter should use `unjsifyRefProp` to convert the JS property to Dart.
+   - The setter should use `jsifyRefProp` to convert the Dart map to JS.
+
+###### Converting Conflicting Function Props
+
+Some function props may conflict with props that are built into the UiProps base class, which get mixed in from `UbiquitousDomPropsMixin`. This will happen if:
+
+1. The component specifies a function that is typically reserved for `UbiquitousDomPropsMixin`. These are typically event handlers on HTML elements, such as `onClick`, `onChange`, etc.
+1. The prop's function signature is not the same as the version on `UbiquitousDomPropsMixin`. This typically takes the form of adding extra parameters that the standard event handlers do not have.
+
+This causes problems in Dart because if the type of the function signature does not match what Dart expects, it will throw a runtime error. Here's what that could look like in TS:
+
+```ts
+interface ArbitraryComponentProps {
+  // note that the `id` parameter isn't usually part of the `onClick` handler signature
+  onClick: (e: React.MouseEvent, id: string) => void;
+}
+```
+
+In Dart, to work around that, you would do something like:
+
+```dart
+@Props(keyNamespace: '')
+mixin ArbitraryComponentProps on UiProps {
+  @Accessor(key: 'onClick')
+  void Function(SyntheticMouseEvent event, String id) onClick_ArbitraryComponent;
+
+  @Deprecated('Use onClick_ArbitraryComponent for proper typing')
+  @override
+  get onClick;
+
+  @Deprecated('Use onClick_ArbitraryComponent for proper typing')
+  @override
+  set onClick(value);
+}
+```
+
+The steps are:
+
+1. Add overridden getters and setters for the prop. Overriding is necessary because `UiProps` already contributes the property.
+1. Deprecate those getters and setters. This isn't necessary to avoid runtime errors, but it's an important step because those properties should not be accessed by devs.
```

Comment:
```suggestion
> This isn't necessary to avoid runtime errors
```
I think the opposite is true. If consumers set callbacks that don't match the signature the JS code calls them with, you'll get a runtime error.